src/main.rs

```diff
@@ -172,10 +205,119 @@ impl ReStreamer {
     pub fn next_frame(&mut self) -> std::io::Result<()> {
         self.file.seek(SeekFrom::Start(self.start))?;
         self.cursor = 0;
+        if self.show_cursor {
+            self.read_input();
+        }
+
         Ok(())
     }
+
+    /// Read input events to figure out pen state and position.
+    fn read_input(&mut self) {
+        let mut down = self.pen_pos.is_some();
+        let mut pos = self.pen_pos.unwrap_or((0, 0));
+        while let Ok(event) = self.input_rx.try_recv() {
+            match event {
+                InputEvent::WacomEvent { event: e } =>
+                    match e {
+                        WacomEvent::InstrumentChange { pen: WacomPen::ToolPen, state: s } => {
+                            down = s;
+                        },
+                        WacomEvent::Hover { position: cgmath::Point2 { x, y }, .. } => {
+                            pos = (x as usize, y as usize);
+                            self.drawing = false;
+                        },
+                        WacomEvent::Draw { .. } => {
+                            // no need to show position while drawing
+                            self.drawing = true;
+                        },
+                        _ => (),
+                    },
+                    _ => (),
+            }
+        }
+
+        self.pen_pos = if down {
+            Some(pos)
+        } else {
+            None
+        };
+    }
+
+    /// Draw pen position into fb data, if necessary (in hover range and not drawing).
+    fn draw_pen_position(&mut self, buf: &mut [u8]) {
+        if let (false, Some((y, x))) = (self.drawing, self.pen_pos) {
+            // we need negative numbers to calculate offsets correctly
+            let width = self.width as isize;
+            let height = self.height as isize;
```

Comment:
```suggestion
            let flip = self.width > self.height;
            let (x, y) = if flip { (y, x) } else { (x, y) };
            // we need negative numbers to calculate offsets correctly
            let width = if flip { self.height } else { self.width } as isize;
            let height = if flip { self.width } else { self.height } as isize;
```
A rm2 without rm2fb has inverted width/height

Reply:
With this change, it works nicely!